extmod/uasyncio/taskgroup.py

```diff
@@ -0,0 +1,244 @@
+# Adapted with permission from the EdgeDB project.
+# Adapted for MicroPython
+
+
+__all__ = ["TaskGroup"]
+
+from . import core
+from . import event
+from . import task
+
+DEBUG = False
+
+
+class TaskGroup:
+    def __init__(self):
+        self._entered = False
+        self._exiting = False
+        self._aborting = False
```

Comment:
Is it possible to combine these 3 into one `self._state` variable?

Reply:
Sure. Will do that.